src/component/state_watcher.rs

```diff
@@ -0,0 +1,27 @@
+use super::Fuselage;
+use std::cell::{Ref, RefCell, RefMut};
+use std::rc::Rc;
+use tokio::sync::Notify;
+
+/// Keeps track of a components model and view.
+///
+/// Borrowing the model and view will notify the component to check for updates.
+#[derive(Debug)]
+pub struct StateWatcher<Component, Widgets> {
+    /// The models and widgets maintained by the component.
+    pub(super) state: RefCell<Fuselage<Component, Widgets>>,
+    pub(super) notifier: Rc<Notify>,
+}
+
+impl<Component, Widgets> StateWatcher<Component, Widgets> {
+    /// Borrows the model and view of a component.
+    pub fn ro(&self) -> Ref<'_, Fuselage<Component, Widgets>> {
```

Comment:
I think `ro` and `rw` are a little to short. It's easy to guess if you're already familiar with Relm4 but for beginners it might be confusing. Something like `get` and `get_mut` would be easier to understand.

Reply:
Changed to `get`